src/components/CallView/Grid/Grid.vue

```diff
@@ -828,6 +929,88 @@ export default {
 			return callParticipantModel.attributes.peerId === this.$store.getters.selectedVideoPeerId
 		},
 
+		isModelWithVideo(callParticipantModel) {
+			return callParticipantModel.attributes.videoAvailable
+				&& (typeof callParticipantModel.attributes.stream === 'object')
+		},
+
+		isModelWithAudio(callParticipantModel) {
+			const participant = this.$store.getters.getParticipantBySessionId(this.token, callParticipantModel.attributes.nextcloudSessionId)
+			if (!participant) {
+				return false
+			}
+			return participant?.permissions & PARTICIPANT.PERMISSIONS.PUBLISH_AUDIO
+		},
+
+		unpromoteSpeaker(model) {
+			const id = model.attributes.nextcloudSessionId
+			// remove the model from the temp promoted speakers
+			const index = this.videosTempPromoted.indexOf(model)
+			if (index === -1) {
+				return
+			}
+
+			this.videosTempPromoted.splice(index, 1)
+
+			if (this.isModelWithVideo(model)) {
+				if (!this.tilesWithVideoMask.includes(id)) {
+					this.tilesWithVideoMask.unshift(id)
+				}
+			} else {
+				if (!this.tilesWithAudioMask.includes(id)) {
+					this.tilesWithAudioMask.unshift(id)
+				}
+			}
```

Comment:
if we added a tile in the mask array during promotion, we already know, where to put it with the next re-render, just need to remove from tempPromoted 
```suggestion
```